src/backend/renderer/gles/mod.rs

```diff
@@ -2363,14 +2363,39 @@ impl<'frame> Frame for GlesFrame<'frame> {
     fn finish(mut self) -> Result<SyncPoint, Self::Error> {
         self.finish_internal()
     }
+
+    fn wait(&mut self, sync: &SyncPoint) {
+        // if the sync point holds a EGLFence we can try
+        // to directly insert it in our context
+        if let Some(fence) = sync.get::<EGLFence>() {
```

Comment:
We need to compare here, that this is the same `EGLDisplay` or does this rely on `fence.wait()` to fail in that case?

Reply:
After reading though it again.. I think we need to compare EGLDisplayHandles here.

Reply:
I was (and I am still) not sure about that. `eglWaitSync` defines
> If display does not match the EGLDisplay passed to eglCreateSync when sync was created, the behavior is undefined. 

which should be fine as this always uses the display it was created from. I would expect it to fail if the current bound context does not support the provided fence.
Can you point me to the paragraph where it demands that the displays has to match the current bound one?

If we need that check I would prefer to extend the `wait` function to receive the current display and check that internally.

Reply:
> which should be fine as this always uses the display it was created from

Yes it is fine, but it is not helpful, if you give the fence to a different renderer (e.g. inside the MultiRenderer).
Because it won't synchronize on the fence, but just insert a server-side wait for the original context:

> The server only blocks execution of commands for the specific context on which eglWaitSync was issued; other contexts implemented by the same server are not affected..

So in those cases, when we call `Renderer::wait` we need to either return an error or try to import the file descriptor from the other context to do the right thing. Otherwise we are not blocking in our current display/context on anything.

Reply:
Also:

>  If no context is current for the currently bound client API (i.e., eglGetCurrentContext returns EGL_NO_CONTEXT), an EGL_BAD_MATCH error is generated. 

So we would want to call `make_current` before.

Reply:
Okay, I changed `EGLFence::wait` to take a display and return an error in case the display is not equal to the display used for creating the fence. `GlesRenderer::wait` now also calls `eglMakeCurrent` before trying to use the fence.